src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/draftorders/upload/UploadDraftOrdersAboutToStartHandler.java

```diff
@@ -0,0 +1,127 @@
+        super(finremCaseDetailsMapper);
+        this.caseAssignedRoleService = caseAssignedRoleService;
+        this.hearingService = hearingService;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_START.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && EventType.DRAFT_ORDERS.equals(eventType);
+    }
+
+    @Override
+    public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(FinremCallbackRequest callbackRequest,
+                                                                              String userAuthorisation) {
+        FinremCaseDetails caseDetails = callbackRequest.getCaseDetails();
+        String caseId = String.valueOf(caseDetails.getId());
+        log.info("Invoking contested {} about to start callback for Case ID: {}", callbackRequest.getEventType(),
+            caseId);
+
+        FinremCaseData finremCaseData = caseDetails.getData();
+        DraftOrdersWrapper draftOrdersWrapper = finremCaseData.getDraftOrdersWrapper();
+        draftOrdersWrapper.setTypeOfDraftOrder(null);
+
+        YesOrNo showUploadPartyQuestion = isShowUploadPartyQuestion(caseId, userAuthorisation)
+            ? YesOrNo.YES : YesOrNo.NO;
+        draftOrdersWrapper.setShowUploadPartyQuestion(showUploadPartyQuestion);
+
+        DynamicMultiSelectList confirmUploadedDocuments = createConfirmUploadDocuments(finremCaseData);
+        DynamicRadioList uploadPartyRadioList = null;
+        if (YesOrNo.YES.equals(showUploadPartyQuestion)) {
+            uploadPartyRadioList = createUploadParty(finremCaseData);
+        }
+
+        draftOrdersWrapper.setUploadSuggestedDraftOrder(UploadSuggestedDraftOrder.builder()
+            .confirmUploadedDocuments(confirmUploadedDocuments)
+            .uploadParty(uploadPartyRadioList)
+            .build());
+
+        draftOrdersWrapper.setUploadAgreedDraftOrder(UploadAgreedDraftOrder.builder()
+            .hearingDetails(hearingService.generateSelectableHearingsAsDynamicList(caseDetails))
+            .confirmUploadedDocuments(confirmUploadedDocuments)
+            .uploadParty(uploadPartyRadioList)
+            .build());
+
+        return GenericAboutToStartOrSubmitCallbackResponse.<FinremCaseData>builder()
+            .data(finremCaseData).build();
+    }
+
+    private DynamicMultiSelectList createConfirmUploadDocuments(FinremCaseData finremCaseData) {
+        String applicantLName = finremCaseData.getApplicantLastName();
+        String respondentLName = finremCaseData.getRespondentLastName();
+
+        DynamicMultiSelectListElement elementConfirmation = DynamicMultiSelectListElement.builder()
+            .label(format("I confirm the uploaded documents are for the %s v %s case", applicantLName, respondentLName))
+            .code("1")
+            .build();
+
```

Comment:
Just checking you don't want to assign "1" to a constant, like you have below

Reply:
Updated